api/src/rest/controllers/rooms.js

```diff
@@ -202,5 +202,53 @@ router.put('/:roomId/join/:userId', async (req, res, next) => {
   }
 });
 
+// PUT leave room (unjoin user and room)
+router.put('/:roomId/leave/:userId', isAuthenticated, async (req, res, next) => {
+  const { roomId, userId } = req.params;
+
+  if (!uuidv4RegExp.test(roomId)) {
+    return next(new ValidationError(
+      'An invalid uuid roomId was passed to leave room route.',
+      'Something went wrong when trying to leave room.'
+    ));
+  }
+
+  if (!uuidv4RegExp.test(userId)) {
+    return next(new ValidationError(
+      'An invalid uuid userId was passed to leave room route.',
+      'Something went wrong when trying to leave room.'
+    ));
+  }
+
+  try {
+    const [result] = await knex(JOIN_USERS_AND_ROOMS)
+      .where({
+        room_id: roomId,
+        user_id: userId,
+      })
+      .returning(['room_id', 'user_id'])
+      .select('*')
+      .del();
+
+    if (!result.room_id) {
```

Comment:
Have you hit these checks yet? I think you may be able to combine this and lines 239 through 244 with something like
```
if (!result || !result.room_id || !result.user_id)
```

Though I do like that you separated them to get more granular error messages, which really does improve the speed of debugging.